src/weights.jl

```diff
@@ -250,6 +250,76 @@ eweights(t::AbstractVector, r::AbstractRange, λ::Real) =
 
 # NOTE: No variance correction is implemented for exponential weights
 
+struct UnitWeights{S<:Real, T<:Real} <: AbstractWeights{S, T, V where V<:Vector{T}}
+    sum::S
+end
+
+UnitWeights{T}(s::S) where {S, T} = UnitWeights{S, T}(s)
+
+@doc """
+    UnitWeights{T}(s)
+
+Construct a `UnitWeights` vector with length `s` and weight elements of type `T`.
+All weight elements are identically one.
+""" UnitWeights
+
+eltype(wv::UnitWeights{S, T})    where {S, T} = T
+values(wv::UnitWeights{S, T})    where {S, T} = fill(one(T), length(wv))
+isempty(wv::UnitWeights{S, T})   where {S, T} = iszero(wv.sum)
+length(wv::UnitWeights{S, T})    where {S, T} = Int(wv.sum)
+Base.size(wv::UnitWeights{S, T}) where {S, T} = Tuple(length(wv))
+
+function Base.getindex(wv::UnitWeights{S, T}, i::Int) where {S, T}
+    (i > 0) & (i <= length(wv)) ? one(T) : Base.throw_boundserror(wv, i)
+end
+
+function Base.getindex(wv::UnitWeights{S, T}, i::AbstractRange{<:Int}) where {S, T}
+    if (minimum(i) > 0) & (maximum(i) <= length(wv))
+        return fill(one(T), length(i))
+    else
+        Base.throw_boundserror(wv, i)
+    end
+end
+
+function Base.getindex(wv::UnitWeights{S, T}, i::AbstractArray{<:Int}) where {S, T}
+    if (minimum(i) > 0) & (maximum(i) <= length(wv))
+        return fill(one(T), size(i))
+    else
+        Base.throw_boundserror(wv, i)
+    end
+end
+
+Base.getindex(wv::UnitWeights{S, T}, ::Colon) where {S, T} = fill(one(T), length(wv))
+
+"""
```

Comment:
I'm not sure we should add this method since it's very different from others: it doesn't accept a vector as input. The constructor may be enough.

Reply:
I lean towards keeping it. One can reasonably expect colon indexes to work on arrays. It works for other weight types too (because they have a `values` subfield).

Reply:
Are we talking about the same thing? I don't see how colon indices are related. :-)

Reply:
Sorry: your comment came just after the definition of `getindex(wv::UnitWeights{S, T}, ::Colon)`, so I assumed that was what it referred to. I don't know which method you have in mind then.

Reply:
OK, let's keep it, but better use a default value for `T` here, and require people to use the constructor to choose a different type: I don't think it should be too common.

The default `T` could be either `Int` or `Bool`. The latter is a bit weird (it gives `true`), but it is sometimes used for a "strong zero" and it always preserves the type with multiplication: `Int8(1) * true === Int8(1)`, which isn't the case with `1`. Not sure whether that's really useful though, since weighted algorithms likely need to widen `Int8` at some point, and for floats `Int` and `Bool` behave the same:`1.0f0 * 1 === 1.0f0`. So maybe the simpler `Int` is better.

Reply:
```
x = uweights(6)
x[:]
type UnitWeights has no fields values
```
vs
```
x[:] == fill(one(1, 6))
```
I think that is good.

Reply:
The default constructor (`uweights(s)`) was already setting the element type to an integer type, since it used the type of `s`, but I made it more explicit in the latest commit.